test/interfaces/multi_particle_distribution.jl

```diff
@@ -0,0 +1,140 @@
+
+using QEDprocesses
+using QEDevents
+using QEDbase
+using Random: Random
+import Random: AbstractRNG, MersenneTwister, default_rng
+
+# only imported, because we want to test,
+# if QEDevents works without this (epsecially the Base.rand which is exported by
+# Distributions)
+using Distributions: Distributions
+
+include("../test_implementation/TestImpl.jl")
+
+RNG = MersenneTwister(137137137)
+
+ATOL = 0.0
+RTOL = sqrt(eps())
+
+struct WrongParticle <: AbstractParticleType end # for type checking in weight
+struct WrongDirection <: ParticleDirection end # for type checking in weight
+
+DIRECTIONS = (Incoming(), Outgoing(), QEDevents.UnknownDirection())
+RND_SEED = ceil(Int, 1e6 * rand(RNG)) # for comparison
+@testset "N=$N" for N in (1, rand(RNG, 2:10))
+    @testset "default properties" begin
+        test_dist_plain = TestImpl.TestMultiParticleDistPlain(N)
+        @test all(
+            QEDevents._particle_direction(test_dist_plain) .== QEDevents.UnknownDirection()
+        )
+        @test QEDevents._momentum_type(test_dist_plain) == SFourMomentum
+    end
+
+    test_particles = Tuple(rand(RNG, TestImpl.PARTICLE_SET, N))
+    test_directions = Tuple(rand(RNG, DIRECTIONS, N))
+    test_dist = TestImpl.TestMultiParticleDist(test_directions, test_particles)
+
+    @testset "static properties" begin
+        @test @inferred QEDevents._particles(test_dist) == test_particles
+        @test @inferred QEDevents._particle_directions(test_dist) == test_directions
+        @test @inferred length(test_dist) == N
+        @test @inferred size(test_dist) == (N,)
+
+        # todo: consider to move _assemble_tuple_types to the test implementation
+        # (groundtruths must not rely on package internals)
```

Comment:
```suggestion
        # todo: consider to move _assemble_tuple_types to the test implementation
        # (groundtruths must not rely on package internals)
        # See https://github.com/QEDjl-project/QEDprocesses.jl/issues/75
```
I think https://github.com/QEDjl-project/QEDprocesses.jl/issues/75 would address this.